### PR TITLE
continuation of PR #8074 Adding spam prevention - fixes #8060

### LINKED ISSUE #8060: Spam prevention - prevent a player from posting too many chat messages in a short time
We want the Tavern, guild, and party chats to have a timed limit preventing people from posting many messages in a row. It should allow only X posts per Y time, with a rolling window, where X and Y are specified as environment variables or constants (constants might be fine because we probably won't want to change this often).

Moderators are exempt. We might also want to exempt Socialites of a high enough level, so perhaps make an exemption for all players who have `contributor.level` greater than perhaps 3 (moderators are level 8). Specify that number as a constant or environment variable.

Any coder who wants to work on this is welcome to post here to claim it, but please do that only if you can start now and finish ideally in a couple of weeks. Please give a progress report after a week so your claim doesn't expire.

FYI @lemoness

### PR DESCRIPTION
**Pull request by TheHollidayInn**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #8060 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

This is a continuation on: https://github.com/HabitRPG/habitica/pull/8074. All credit goes to that author: 9c877aed-a20b-4aa0-b4b3-85e4456f934e

[//]: # (Put User ID in here - found in Settings -> API)

----
UUID:

### CODE REVIEW

**Review by SabreCat — APPROVED** [review]

File: website/common/locales/en/messages.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
   "messageGroupChatFlagAlreadyReported": "You have already reported this message",
   "messageGroupChatNotFound": "Message not found!",
   "messageGroupChatAdminClearFlagCount": "Only an admin can clear the flag count!",
+  "messageGroupChatSpam": "Whoops, looks like you're posting too many messages! Please wait a minute and try again. The Tavern chat only holds 200 messages at a time, so Habitica encourages posting longer, more thoughtful messages and consolidating replies. Can't wait to hear what you have to say :)",
```
Comment (khipkin) [inline website/common/locales/en/messages.json]: Nit: end last sentence with a period

File: website/server/models/group.js [hunk 2]
```diff
@@ -46,6 +46,16 @@ const CRON_SAFE_MODE = nconf.get('CRON_SAFE_MODE') === 'true';
 const CRON_SEMI_SAFE_MODE = nconf.get('CRON_SEMI_SAFE_MODE') === 'true';
 const MAX_UPDATE_RETRIES = 5;
 
+/*
+#  Spam constants to limit people from sending too many messages too quickly
+#    MESSAGE_LIMIT - The amount of messages that can be sent in a time window
```
Comment (khipkin) [inline website/server/models/group.js]: nit: update comments so var names match below

File: website/common/locales/en/messages.json [hunk 3]
```diff
@@ -56,7 +56,7 @@
   "messageGroupChatFlagAlreadyReported": "You have already reported this message",
   "messageGroupChatNotFound": "Message not found!",
   "messageGroupChatAdminClearFlagCount": "Only an admin can clear the flag count!",
-  "messageGroupChatSpam": "Whoops, looks like you're posting too many messages! Please wait a minute and try again. The Tavern chat only holds 200 messages at a time, so Habitica encourages posting longer, more thoughtful messages and consolidating replies. Can't wait to hear what you have to say :)",
+  "messageGroupChatSpam": "Whoops, looks like you're posting too many messages! Please wait a minute and try again. The Tavern chat only holds 200 messages at a time, so Habitica encourages posting longer, more thoughtful messages and consolidating replies. Can't wait to hear what you have to say :).",
```
Comment (khipkin) [inline website/common/locales/en/messages.json]: I meant before the smiley face. 😛

Comment (TheHollidayInn) [reply website/common/locales/en/messages.json]: Ah, I guess I thought the smily face counted as part of the sentence. I'm unsure about how emojis work in syntax :P

**Pull request merged**